components/base/kinematicbase/ptgKinematics.go

```diff
@@ -128,6 +146,10 @@ func (ptgk *ptgBaseKinematics) GoToInputs(ctx context.Context, inputs []referenc
 		return multierr.Combine(err, ptgk.Base.Stop(ctx, nil))
 	}
 
+	ptgk.inputLock.Lock()
+	ptgk.currentInput = []referenceframe.Input{inputs[0], inputs[1], {0}}
```

Comment:
I thought I replied to this last week and apparently did not hit submit.

If we want `CurrentInputs()` to return a point we are guaranteed to have hit, rather than a point we are currently going to, then we need to do this once before the loop, and then thereafter update after the SetVelocity call returns.

I could combine them, yes, but I'd just be trading for cacheing the trajectory nodes in the loop in a variable to assign at the start of the next iteration. IMO it's not any more efficient or less complicated than what's already here.

Reply:
I thought it made sense from a simplicity/readability perspective but yeah the output would be the same

Reply:
I implemented what you suggested anyway to see what it would look like and actually like the look of it way better. So will do this.